src/components/UploadMetadata.tsx

```diff
@@ -0,0 +1,207 @@
+import { FC, useState } from 'react';
+import { WebBundlr } from '@bundlr-network/client';
+import { LAMPORTS_PER_SOL } from '@solana/web3.js';
+import { useWallet, useConnection } from '@solana/wallet-adapter-react';
+import { PhantomWalletAdapter } from '@solana/wallet-adapter-phantom';
+
+import { notify } from '../utils/notifications';
+
+export const UploadMetadata: FC = ({}) => {
+  const wallet = useWallet();
+  const { connection } = useConnection();
+  const [provider, setProvider] = useState(null);
+  const [address, setAddress] = useState(null);
+  const [bundlr, setBundlr] = useState(null);
+  const [imageFile, setImageFile] = useState(null);
+  const [imageUrl, setImageUrl] = useState(null);
+  const [metadata, setMetadata] = useState(null);
+  const [metadataUrl, setMetadataUrl] = useState(null);
+
+  const initializeProvider = async () => {
+    if (window.solana.isPhantom) {
+      await window.solana.connect();
+      const provider = new PhantomWalletAdapter();
+      await provider.connect();
+      setProvider(provider);
+      notify({ type: 'success', message: `Provider: ${provider.name}` });
+    }
+  };
+
+  const initializeBundlr = async () => {
+    // initialise a bundlr client
+    const bundler = new WebBundlr(
+      'https://devnet.bundlr.network',
```

Comment:
I agree. Now that I know it works, I can make adjustments.

Reply:
Update on this: Phantom's adapter doesn't track the network you're using, and Solflare does have a network property but it's not public so it can't be used. It's also defaulted to mainnet-beta, even though when I created my Solflare wallet I set it devnet. Bundlr also expects the URL to the RPC (if you're using Bundlr's `devnet`) and not just the name. The only way I can see being able to set this is using a `select` field to show the mainnet and devnet options and connecting based off user selection.

Reply:
Completed.